packages/patternfly-4/react-core/src/components/Select/Select.js

```diff
@@ -0,0 +1,141 @@
+  /** Flag indicating selection is valid */
+  isValid: PropTypes.bool,
+  /** Flag indicating the Select is disabled */
+  isDisabled: PropTypes.bool,
+  /** Optional callback for updating when selection loses focus */
+  onBlur: PropTypes.func,
+  /** Optional callback for updating when selection gets focus */
+  onFocus: PropTypes.func,
+  /** Optional callback for updating when selection changes */
+  onChange: PropTypes.func,
+  /** Custom flag to show that the Select requires an associated id or aria-label. */
+  'aria-label': props => {
+    if (!props.id && !props['aria-label']) {
+      return new Error('Select requires either an id or aria-label to be specified');
+    }
+    return null;
+  }
+};
+
+const defaultProps = {
+  className: '',
+  value: '',
+  options: null,
+  isOptionsGrouped: false,
+  getOptionGroupLabel: () => undefined,
+  getOptionFromGroup: () => undefined,
+  getOptionLabel: () => undefined,
+  getOptionValue: () => undefined,
+  getOptionDisabled: () => undefined,
+  isValid: true,
+  isDisabled: false,
+  onBlur: () => undefined,
+  onFocus: () => undefined,
+  onChange: () => undefined,
+  'aria-label': null
+};
+
+class Select extends React.Component {
+  /** call callback to handle selection change */
+  handleChange = event => {
+    this.props.onChange(event);
+  };
+
+  /** call callback to handle selection change */
+  handleBlur = event => {
+    this.props.onBlur(event);
+  };
+
+  /** call callback to handle selection change */
+  handleFocus = event => {
+    this.props.onFocus(event);
+  };
+
+  /* returns the options */
+  getOptions = options => {
+    const { getOptionDisabled, getOptionValue, getOptionLabel } = this.props;
+    return options
+      ? options.map(
+          (option, index) =>
+            getOptionDisabled(option) ? (
+              <option disabled key={index} value={getOptionValue(option)}>
+                {getOptionLabel(option)}
+              </option>
+            ) : (
+              <option key={index} value={getOptionValue(option)}>
+                {getOptionLabel(option)}
+              </option>
+            )
+        )
+      : '';
+  };
+
+  render() {
+    const {
+      className,
+      options,
+      value,
+      isOptionsGrouped,
+      getOptionGroupLabel,
+      getOptionFromGroup,
+      getOptionDisabled,
+      getOptionValue,
+      getOptionLabel,
+      isValid,
+      isDisabled,
+      onBlur,
+      onFocus,
+      ...props
+    } = this.props;
+    return (
+      <select
+        {...props}
+        className={css(styles.formControl, className, !isValid && styles.modifiers.invalid)}
+        aria-invalid={!isValid}
+        onChange={this.handleChange}
+        onBlur={onBlur}
+        onFocus={onFocus}
+        disabled={isDisabled}
+        value={value}
+      >
+        {options &&
+          (isOptionsGrouped
+            ? options.map((group, index) => (
+                <optgroup key={index} label={getOptionGroupLabel(group)}>
```

Comment:
A better way to do this might be to do something like
```
[
  { label: 'Foor', value: 'bar' },
  { 
     title: 'Option Group Label',
     items: [{ label: 'Baz, value: 'another value' }]
   }
]
```
For options. This would allow the consumer to control the groups better instead of needing switch statements or something else to handle getOptionsGroupLabel.

Reply:
Again, are we making assumptions about the options object?

Reply:
I think the lib can set sane defaults of label, value, is disabled, items, etc for the structure. If something more advanced is needed call backs can be added. These are just suggestions though. As I am no longer a maintainer, it will be up to RH and the team for what needs are needed. My suggestions puts PF more inline with other larger component libs.